src/main/scala/zio/s3/errors.scala

```diff
@@ -0,0 +1,28 @@
+/*
+ * Copyright 2017-2020 John A. De Goes and the ZIO Contributors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package zio.s3
+
+import java.io.IOException
+
+sealed trait S3Failure
```

Comment:
with
```
sealed trait S3Failure {
  def message: String

  def cause: Throwable
}
```
I can improve my errors with `cause`:
```
def s3Exception(cause: S3Exception) = new RuntimeException(s"s3Exception: $cause", cause)
```

Reply:
S3Exception dont extend S3Failure, 
im only exposing the error from amazon sdk `software.amazon.awssdk.services.s3.model.S3Exception`

i should rename S3Failure  to S3ConnectionFailure to remove any ambiguity

Reply:
sorry, wrong copied line, should be
```
    def s3Err(e: S3Failure) = new RuntimeException(s"s3Err: $e", e.cause)
```